Newsletters: add 295 (2024-03-27) (#1571)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2024-03-27-newsletter.md

  for each additional sponsorship.

    After hearing about Habovštiak's idea, David Harding
    [posted][harding sponsor] to Delving Bitcoin an efficiency
    improvement he and Rubin had previously developed in January.  The
    sponsor transaction commits to the boosted transaction using the
    signature commitment message, which is never published onchain, so
    zero block space is used for a single commitment.  To allow this,
    the sponsor transaction must appear in blocks and [package
    relay][topic package relay] messages immediately after the boosted
    transaction, allowing full node verifiers to infer the txid of the
    boosted transaction when they verify the sponsor transaction.

    For cases where a block may contain multiple sponsor transactions
    that each commit to some of the same boosted transactions, it's not
    possible to simply have a series of boosted transactions appear
    immediately before their sponsors, so entirely inferable commitments
    is not an option.  Harding describes a simple alternative that only
    uses 0.5 vbytes per boosted transaction; Anthony Towns
    [improves][towns sponsor] upon that with a version that would never

>     use more then 0.5 vbytes per boost and would use less space

Reviewer: ```suggestion
    use more than 0.5 vbytes per boost and would use less space
```

    transaction validity caching, and concludes with a table showing the
    relative block space consumed by various current and proposed fee
    bumping techniques.  At 0.5 vbytes or less per boost, the most
    efficient form of fee sponsorship is only bested by the 0.0 vbytes
    used in the best case with RBF and paying miners [out-of-band][topic
    out-of-band fees].  Because fee sponsorship allows dynamic fee
    bumping and is almost as efficient as paying miners out-of-band, it
    may resolve a major concern with protocols that depend on [exogenous
    fees][topic fee sourcing].

- **Mempool-based feerate estimation:** Abubakar Sadiq Ismail
  [posted][ismail fees] to Delving Bitcoin about improving Bitcoin
  Core's [feerate estimation][topic fee estimation] using data from a
  node's local mempool.  Currently, Bitcoin Core generates estimates by
  recording the block height when each unconfirmed transaction is
  received, the block height when it is confirmed, and its feerate.
  When all of that information is known, the delta between received
  height and confirmed height is used to update an exponentially

>   weighted moving average for a bucket that represents range of

Reviewer: ```suggestion
  weighted moving average for a bucket that represents a range of
```

  feerates.  For example, a transaction that takes 100 blocks to confirm
  with a feerate of 1.1 sat/vbyte will be incorporated into the average
  for the 1 sat/vbyte bucket.

  An advantage of this approach is its resistance to manipulation: all
  transactions must be both relayed (meaning they're available to all
  miners) and confirmed (meaning they can't violate any consensus
  rules).  A disadvantage is that it only updates once per block and
  can lag far behind other estimates that use real-time mempool
  information.

  Ismail has taken [previous discussion][bitcoin core #27995] about
  incorporating mempool data into feerate estimates, written some
  preliminary code, and performed an analysis showing how the current
  algorithm and a new algorithm compare (not including some safety
  checks).  A [reply][harding fees] to the thread also linked to
  [previous research][alm fees] on this topic by Kalle Alm and led to a
  discussion about whether mempool information should be used to both

>   raiser and lower feerate estimates, or if it should only be used to

Reviewer: ```suggestion
  raise and lower feerate estimates, or if it should only be used to
```

  the estimates more useful; the advantage of only lowering estimates
  using mempool data (while only raising estimates using the existing
  confirmation-based estimation) is that it could be more resistant to
  manipulation and positive feedback loops.

  Discussion was ongoing as of this writing.

## Selected Q&A from Bitcoin Stack Exchange

*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
contributors look for answers to their questions---or when we have a
few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aa
nswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

FIXME:bitschmidty

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [Bitcoin Core 26.1rc2][] is a release candidate for a maintenance release
  of the network's predominant full node implementation.

> - [Bitcoin Core 27.0rc1][] is a release candidate for the next major

Reviewer 2: Probably on your radar, but: https://github.com/bitcoin-core/bitcoin-devwiki/wiki/27.0-Release-Candidate-Testing-Guide

(@davidgumberg joining the Recap week)

  lower estimates.  The advantage of doing both is that it overall makes
  the estimates more useful; the advantage of only lowering estimates
  using mempool data (while only raising estimates using the existing
  confirmation-based estimation) is that it could be more resistant to
  manipulation and positive feedback loops.

  Discussion was ongoing as of this writing.

## Selected Q&A from Bitcoin Stack Exchange

*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
contributors look for answers to their questions---or when we have a
few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aa
nswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [What are the risks of running a pre-SegWit node (0.12.1)?]({{bse}}122211)
  Michael Folkson, Vojtěch Strnad, and Murch list downsides to running Bitcoin
  Core 0.12.1 for an individual user including a higher risk of accepting an

>   invalid transaction or block, increased vulnerability to double spend attacks,

Reviewer: ```suggestion
  invalid transaction or block, increased vulnerability to double-spend attacks,
```

  higher reliance on others to do updated consensus validation, much slower
  block validation, missing many performance improvements, inability to use
  [compact block relay][topic compact block relay], not relaying ~95% of current
  unconfirmed transactions, less accurate [fee estimation][topic fee
  estimation], and vulnerability to security issues fixed in previous versions.
  Wallet users of 0.12.1 would also miss out on developments around
  [miniscript][topic miniscript], [descriptor][topic descriptors] wallets, and
  the fee savings and additional script capabilities enabled by [segwit][topic
  segwit], [taproot][topic taproot], and [schnorr signatures][topic schnorr
  signatures]. Effects on the Bitcoin network if Bitcoin Core

>   0.12.1 was more broadly adopted could include: higher chance of invalid

Reviewer: ```suggestion
  0.12.1 was more broadly adopted could include: a greater chance of invalid
```

layout: newsletter
lang: en
---
This week's newsletter announces the disclosure of a bandwidth-wasting
attack affecting Bitcoin Core and related nodes, describes several
improvements to the idea for transaction fee sponsorship, and summarizes
a discussion about using live mempool data to improve Bitcoin Core's
feerate estimation.  Also included are our regular sections with
selected questions and answers from the Bitcoin Stack Exchange,
announcements of new releases and release candidates, and notable
changes to popular Bitcoin infrastructure projects.

## News

- **Disclosure of free relay attack:** a bandwidth-wasting attack was

>   [described][topic responsible disclosures] to the Bitcoin-Dev mailing

Reviewer 3: Shouldn't it be the link below?
https://gnusha.org/pi/bitcoindev/Zfg%2F6IZyA%2FiInyMx@petertodd.org/

Author: Thanks!!!!

nswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [What are the risks of running a pre-SegWit node (0.12.1)?]({{bse}}122211)

>   Michael Folkson, Vojtěch Strnad, and Murch list downsides to running Bitcoin

Reviewer: ```suggestion
  Michael Folkson, Vojtěch Strnad, and Murch list the downsides to running Bitcoin
```

Reviewer 4: IMO was fine before.

>   Core 0.12.1 for an individual user including higher risk of accepting an

Reviewer: ```suggestion
  Core 0.12.1 for an individual user including a higher risk of accepting an
```

Reviewer 4: IMO was fine before.

  invalid transaction or block, increased vulnerability to double spend attacks,
  higher reliance on others to do updated consensus validation, much slower
  block validation, missing many performance improvements, inability to use
  [compact block relay][topic compact block relay], not relaying ~95% of current
  unconfirmed transactions, less accurate [fee estimation][topic fee
  estimation], and vulnerability to security issues fixed in previous versions.

>   Wallet users of 0.12.1 would also miss out in developments around

Reviewer: ```suggestion
  Wallet users of 0.12.1 would also miss out on developments around
```

  [miniscript][topic miniscript], [descriptor][topic descriptors] wallets, and
  the fee savings and additional script capabilities enabled by [segwit][topic
  segwit], [taproot][topic taproot], and [schnorr signatures][topic schnorr
  signatures]. Effects on the Bitcoin network if Bitcoin Core

>   0.12.0 were more broadly adopted could include: higher chance of invalid

Reviewer: ```suggestion
  0.12.0 was more broadly adopted could include a higher chance of invalid
```

Reviewer 4: IMO was fine before.

Reviewer: What is being more broadly adopted is Bitcoin Core 0.12.0, which is singular. This word I'm suggesting (were, was) applies to "Bitcoin Core 0.12.0", not to "effects". So "was" is the correct word to use here.

  blocks being accepted by the network and associated reorg risk, miner
  centralization pressure from increased stale block risk, and decreased mining
  rewards for miners running that version.

- [When is OP_RETURN cheaper than OP_FALSE OP_IF?]({{bse}}122321)
  Vojtěch Strnad details the overheads associated with `OP_RETURN`-based data
  embedding and `OP_FALSE` `OP_IF`-based embedding, concluding that "`OP_RETURN`
  is cheaper for data smaller than 143 bytes".

- [Why does BIP-340 use secp256k1?]({{bse}}122268)
  Pieter Wuille explains the rationale of choosing secp256k1 over Ed25519 for
  [BIP340][] schnorr signatures and notes "reusability of existing key derivation
  infrastructure" and no "changing security assumptions" as reasons for the choice.

- [What criteria does Bitcoin Core use to create block templates?]({{bse}}122216)

>   Murch explains Bitcoin Core's current ancestor set feerate based algorithm for

Reviewer: ```suggestion
  Murch explains Bitcoin Core's current ancestor set feerate-based algorithm for
```

  transaction selection for a block candidate and mentions ongoing work on [cluster
  mempool][topic cluster mempool] which offers various improvements.

- [How does the initialblockdownload field in the getblockchaininfo RPC work?]({{bse}}122169)

>   Pieter Wuille notes the two conditions that need to occur after node startup,
>   in order for `initialblockdownload` to become false:

Reviewer: ```suggestion
  Pieter Wuille notes the two conditions that need to occur after node startup
  for `initialblockdownload` to become false:
```
